hyperspy/_signals/eds_tem.py

```diff
@@ -242,3 +245,171 @@ def get_calibration_from(self, ref, nb_pix=1):
+        kwargs
+            The extra keyword arguments for plotting. See
+            `utils.plot.plot_signals`
+
+        Return
+        ------
+        A list of quantified elemental maps giving the composition of the
+        sample in weight or atomic percent.
+
+        Examples
+        ---------
+        >>> #s is a signals.EDSTEMSpectrum
+        >>> s.set_elements(["Al", "Cr", "Ni"])
+        >>> s.add_lines()
+        >>> kfactors = [0.982, 1.32, 1.60]
+        >>> intensities = s.get_lines_intensity()
+        >>> res = s.quantification_cliff_lorimer(intensities,kfactors)
+
+        See also
+        --------
+        vacuum_mask
+        """
+        if isinstance(navigation_mask, float):
+            navigation_mask = self.vacuum_mask(navigation_mask, closing).data
+        elif navigation_mask is not None:
+            navigation_mask = navigation_mask.data
+        xray_lines = self.metadata.Sample.xray_lines
+        composition = utils.stack(intensities)
+        composition.data = utils_eds.quantification_cliff_lorimer(
+            composition.data, kfactors=kfactors,
+            mask=navigation_mask) * 100.
+        composition = composition.split()
+        if composition_units == 'atomic':
+            composition = utils.material.weight_to_atomic(composition)
+        for i, xray_line in enumerate(xray_lines):
+            element, line = utils_eds._get_element_and_line(xray_line)
+            composition[i].metadata.General.title = composition_units + \
+                ' percent of ' + element
+            composition[i].metadata.set_item("Sample.elements", ([element]))
+            composition[i].metadata.set_item(
+                "Sample.xray_lines", ([xray_line]))
+            if plot_result and \
+                    composition[i].axes_manager.signal_dimension == 0:
+                print("%s (%s): Composition = %.2f %s percent"
+                      % (element, xray_line, composition[i].data,
+                         composition_units))
+        if plot_result and composition[i].axes_manager.signal_dimension != 0:
+            utils.plot.plot_signals(composition, **kwargs)
+        return composition
+
+    def vacuum_mask(self, threshold=1.0, closing=True):
+        """
+        Generate mask of the vacuum region
+
+        Parameters
+        ----------
+        threshold: float
+            For a given pixel, maximum value in the energy axis below which the
+            pixel is considered as vacuum.
+        closing: bool
+            If true, applied a morphologic closing to the mask
+
+        Return
+        ------
+        mask: signal
+            The mask of the region
+        """
+        from scipy.ndimage.morphology import binary_dilation, binary_erosion
+        mask = (self.max(-1) <= threshold)
+        if closing:
+            mask.data = binary_dilation(mask.data, border_value=0)
+            mask.data = binary_erosion(mask.data, border_value=1)
```

Comment:
This closing only gets rid of `False` holes in the mask (i.e. a single `False` pixel, surrounded by `True` on most sides). Is it intended to leave the `True` holes? Two more reversed identical operations would get rid of them (i.e. first dilation, then erosion, so that the whole treatment would look like dilation -> erosion -> erosion -> dilation)